oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/IndexController.java

```diff
@@ -91,12 +118,42 @@ public static String getPhysicalTableName(String logicName) {
             return Optional.ofNullable(LOGIC_INDICES_CATALOG.get(logicName)).orElse(logicName);
         }
 
-        public static void registerRelation(String logicName, String physicalName) {
-            LOGIC_INDICES_CATALOG.put(logicName, physicalName);
+        public static void registerRelation(Model model, String physicalName) {
```

Comment:
Is this method going to face concurrency cases? I notice you are using `ConcurrentHashMap`.

Notice, even if it is, you face a data loss issue when calling a closure in a `computeIfAbsent` concurrently.

Although, I think this is no concurrency case, right?

Reply:
fixed